jumanji/environments/routing/connector/generator.py

```diff
@@ -106,3 +125,444 @@ def __call__(self, key: chex.PRNGKey) -> State:
         step_count = jnp.array(0, jnp.int32)
 
         return State(key=key, grid=grid, step_count=step_count, agents=agents)
+
+
+class RandomWalkGenerator(Generator):
+    """Randomly generates `Connector` grids that are guaranteed be solvable. This generator places
+    start positions randomly on the grid and performs a random walk from each.  Targets are placed
+    at their terminuses.
+    """
+
+    def __init__(self, grid_size: int, num_agents: int) -> None:
+        """Instantiates a `ParallelRandomWalkGenerator.
+
+        Args:
+            grid_size: size of the square grid to generate.
+            num_agents: number of agents/paths on the grid.
+        """
+        super().__init__(grid_size, num_agents)
+        self.cols = grid_size
+        self.rows = grid_size
+
+    def __call__(self, key: chex.PRNGKey) -> State:
+        """Generates a `Connector` state that contains the grid and the agents' layout.
+
+        Returns:
+            A `Connector` state.
+        """
+        key, _ = jax.random.split(key)
+
+        grid = jnp.zeros((self.grid_size, self.grid_size), dtype=jnp.int32)
+        starts, targets, _ = self.generate_board(key)
+        starts = tuple(starts)
+        targets = tuple(targets)
+        agent_position_values = jax.vmap(get_position)(jnp.arange(self.num_agents))
+        agent_target_values = jax.vmap(get_target)(jnp.arange(self.num_agents))
+
+        # Transpose the agent_position_values to match the shape of the grid.
+        # Place the agent values at starts and targets.
+        grid = grid.at[starts].set(agent_position_values)
+        grid = grid.at[targets].set(agent_target_values)
+
+        # Create the agent pytree that corresponds to the grid.
+
+        agents = jax.vmap(Agent)(
+            id=jnp.arange(self.num_agents),
+            start=jnp.stack(starts, axis=1),
+            target=jnp.stack(targets, axis=1),
+            position=jnp.stack(starts, axis=1),
+        )
+
+        step_count = jnp.array(0, jnp.int32)
+
+        return State(key=key, grid=grid, step_count=step_count, agents=agents)
+
+    def generate_board(
+        self, key: chex.PRNGKey
+    ) -> Tuple[chex.Array, chex.Array, chex.Array]:
+        """Generates solvable board using random walk.
+
+        Args:
+            key: random key.
+
+        Returns:
+            Tuple containing head and targets positions for each wire and the solved board
+            generated in the random walk.
+        """
+        grid = self._return_blank_board()
```

Comment:
I believe we do not need a method for this. We could directly use `grid = jnp.zeros((self.rows, self.cols), dtype=jnp.int32)`